plugins/legacy-plugin-chart-calendar/src/vendor/cal-heatmap.js

```diff
@@ -1652,6 +1652,20 @@ CalHeatMap.prototype = {
       }
     }
 
+    function formatTextFill(value) {
```

Comment:
This is super nifty! I wonder if we should label it as something more descriptive, like "getContrastingColor" so that we can eventually (after monorepo migration, perhaps) export it from a shared module, like Superset's `colorUtils.ts` file.